packages/augur-core/source/contracts/Cash.sol

```diff
@@ -3,42 +3,156 @@ pragma solidity 0.5.4;
+        internalTransfer(msg.sender, _to, _amount);
+        return true;
+    }
+
+    function transferFrom(address _from, address _to, uint256 _amount) public returns (bool) {
+        uint256 _allowance = allowed[_from][msg.sender];
+        require(_amount <= _allowance, "Not enough funds allowed");
+
+        if (_allowance != ETERNAL_APPROVAL_VALUE) {
+            allowed[_from][msg.sender] = _allowance.sub(_amount);
+        }
+
+        internalTransfer(_from, _to, _amount);
+        return true;
+    }
+
+    function internalTransfer(address _from, address _to, uint256 _amount) internal returns (bool) {
+        require(_to != address(0), "Cannot send to 0x0");
+        require(balances[_from] >= _amount, "SEND Not enough funds");
+
+        balances[_from] = balances[_from].sub(_amount);
+        balances[_to] = balances[_to].add(_amount);
+        emit Transfer(_from, _to, _amount);
+        return true;
+    }
+
+    function totalSupply() public view returns (uint256) {
+        return supply;
+    }
+
+    function balanceOf(address _owner) public view returns (uint256) {
+        return balances[_owner];
+    }
+
+    function approve(address _spender, uint256 _amount) public returns (bool) {
+        approveInternal(msg.sender, _spender, _amount);
+        return true;
+    }
+
+    function increaseApproval(address _spender, uint _addedValue) public returns (bool) {
+        approveInternal(msg.sender, _spender, allowed[msg.sender][_spender].add(_addedValue));
+        return true;
+    }
+
+    function decreaseApproval(address _spender, uint _subtractedValue) public returns (bool) {
+        uint oldValue = allowed[msg.sender][_spender];
+        if (_subtractedValue > oldValue) {
+            approveInternal(msg.sender, _spender, 0);
+        } else {
+            approveInternal(msg.sender, _spender, oldValue.sub(_subtractedValue));
+        }
+        return true;
+    }
+
+    function approveInternal(address _owner, address _spender, uint256 _allowance) internal returns (bool) {
+        allowed[_owner][_spender] = _allowance;
+        emit Approval(_owner, _spender, _allowance);
         return true;
     }
 
+    function allowance(address _owner, address _spender) public view returns (uint256) {
+        return allowed[_owner][_spender];
+    }
+
     function faucet(uint256 _amount) public returns (bool) {
+        daiVat.faucet(address(daiJoin), _amount * DAI_ONE);
         mint(msg.sender, _amount);
         return true;
     }
 
-    function getTypeName() public view returns (bytes32) {
-        return "Cash";
+    function sub(uint x, uint y) internal pure returns (uint z) {
+        require((z = x - y) <= x, "math-sub-underflow");
     }
 
-    function onMint(address, uint256) internal returns (bool) {
-        return true;
+    function joinMint(address usr, uint wad) public auth returns (bool) {
+        return mint(usr, wad);
+    }
+
+    function joinBurn(address usr, uint wad) public returns (bool) {
```

Comment:
These two functions don't follow variable naming convention.  Both in their word choice and in the `_` prefix.